cloudinit/distros/tests/test_networking.py

```diff
@@ -126,6 +126,22 @@ class TestGenerateFallbackConfig:
     def test_generate_fallback_config_default_v2(self, generic_networking_cls):
         pass
 
+class TestBSDGenerateFallbackConfig:
+    def test_generate_fallback_config_default_v2(self, two_physdevs_netcfg):
+        networking = BSDNetworking()
+        with mock.patch.object(
+            networking, "find_fallback_nic"
+        ) as m_find_fallback_nic:
+            m_find_fallback_nic.side_effect = 'eth0'
```

Comment:
I think this should be return_value not side_effect (which is generally for callables and exceptions etc which get exec'd each time the find_fallback_nic func is called)